internal/saucecloud/cloud.go

```diff
@@ -336,11 +340,31 @@ func (r *CloudRunner) runJobs(jobOpts chan job.StartOptions, results chan<- resu
 			continue
 		}
 
+		// Attempt starts from 0 while MaxAttempts starts from 1. So here is comparing Attempt with MaxAttempts-1
+		if opts.Attempt < opts.MaxAttempts-1 && opts.CurrentPassCount < opts.PassCount {
+			opts.Attempt++
+			jobOpts <- opts
+			log.Info().Str("suite", opts.DisplayName).Str("attempt", fmt.Sprintf("%d of %d", opts.Attempt+1, opts.MaxAttempts)).Msg("Retrying suite.")
+			continue
+		}
+
 		if r.FailFast && !jobData.Passed {
 			log.Warn().Err(err).Msg("FailFast mode enabled. Skipping upcoming suites.")
 			r.interrupted = true
 		}
 
+		if opts.MaxAttempts != 0 {
+			if opts.CurrentPassCount >= opts.PassCount {
+				log.Info().Str("suite", opts.DisplayName).Msg("Passed threshold")
+				jobData.Status = job.StatePassed
+				jobData.Passed = true
+			} else {
+				log.Info().Str("suite", opts.DisplayName).Msg("Failed to pass threshold")
```

Comment:
Error or warning more appropriate here. I chose 'error' since that's what we use for indicating failed suites as well.

```suggestion
				log.Error().Str("suite", opts.DisplayName).Msg("Failed to pass threshold")
```